docs/auth/oidc.md

```diff
@@ -0,0 +1,280 @@
+      ...defaultAuthProviderFactories,
+     'my-auth-provider': providers.oidc.create({
++       signIn: {
++         resolver(info, ctx) {
++           const userRef = stringifyEntityRef({
++             kind: 'User',
++             name: info.profile.email!,
++             namespace: DEFAULT_NAMESPACE,
++           });
++           return ctx.issueToken({
++             claims: {
++               sub: userRef, // The user's own identity
++               ent: [userRef], // A list of identities that the user claims ownership through
++             },
++           });
++         },
++       },
+     }),
+    }
+```
+
+We could replace the whole `signIn` object with
+`providers.microsoft.resolvers.emailLocalPartMatchingUserEntityName() as any` and as long
+as you have the correct user configured in the catalog it should work, note the `as any`
+part, TS will error out because of typings, so we type cast it to any, it should still
+work given the response of an OIDC or OAuth2 request to microsoft is very similar.
+
+### The configuration
+
+Since we are using our custom OIDC Auth Provider, we need to add a configuration based
+on the provider used, in this case based on OIDC protocol (remember the 3rd party has to
+support the protocol).
+
+In this example we'll configure OIDC with `my-auth-provider`, to do so we need to
+[Create app registration][2] in the Azure console, the only difference is that the
+`http://localhost:7007/api/auth/microsoft/handler/frame` URL needs to change to
+`http://localhost:7007/api/auth/my-auth-provider/handler/frame`.
+
+Then we need to configure the env variables for the provider, based on the provider's code
+in `plugins/auth-backend/src/providers/oidc/provider.ts` we need the following variables
+in the `app-config.yaml`:
+
+```yaml
+auth:
+  environment: development
+  ### Providing an auth.session.secret will enable session support in the auth-backend
+  session:
+    secret: ${SESSION_SECRET}
+  providers:
+    my-auth-provider:
+      # Note that you must define a session secret (see above) since the oidc provider requires session support.
+      # Note that by default, this provider will use the 'none' prompt which assumes that your are already logged on in the IDP.
+      # You should set prompt to:
+      # - auto: will let the IDP decide if you need to log on or if you can skip login when you have an active SSO session
+      # - login: will force the IDP to always present a login form to the user
+      development:
+        metadataUrl: ${AUTH_OIDC_METADATA_URL}
+        clientId: ${AUTH_OIDC_CLIENT_ID}
+        clientSecret: ${AUTH_OIDC_CLIENT_SECRET}
+        authorizationUrl: ${AUTH_OIDC_AUTH_URL}
+        tokenUrl: ${AUTH_OIDC_TOKEN_URL}
+        tokenSignedResponseAlg: ${AUTH_OIDC_TOKEN_SIGNED_RESPONSE_ALG} # default='RS256'
+        scope: ${AUTH_OIDC_SCOPE} # default='openid profile email'
+        prompt: ${AUTH_OIDC_PROMPT} # default=none (allowed values: auto, none, consent, login)
```

Comment:
At the very least we should keep `metadataUrl`, `clientId`, `clientSecret`, `authorizationUrl`, `tokenUrl`, and `prompt`, I think this example will not run without these.

Reply:
I think that's going to be highly dependent on the IdP. Everyone implements OAuth 2 and OIDC differently. These 3 are the ones that are required by the auth provider implementation itself

Reply:
So these are required only if we're using Azure OIDC?

Reply:
The metadata endpoint on an OIDC authority will [describe](https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata) its authorization and token endpoints, allowing clients to discover them instead of being statically configured